audit/audit.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2018 Bluzelle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#include <audit/audit.hpp>
+#include <boost/beast/core/detail/base64.hpp>
+#include <boost/format.hpp>
+
+using namespace bzn;
+
+audit::audit(std::shared_ptr<bzn::node_base> node): node(node)
+{
```

Comment:
I'd format this as:

audit::audit(...)
    : node(node)
{
}
a bit easier to read

Reply:
please also use std::move(node)